fedimint-client/src/transaction/sm.rs

```diff
@@ -39,44 +34,28 @@ impl IntoDynInstance for TxSubmissionContext {
     }
 }
 
-// TODO: refactor states into their own structs that impl `State`. The enum
-// merely dispatches fn calls to the right state impl in that scenario
 #[cfg_attr(doc, aquamarine::aquamarine)]
-/// State machine to (re-)submit a transaction till it is either confirmed or
+/// State machine to (re-)submit a transaction until it is either accepted or
 /// rejected by the federation
 ///
 /// ```mermaid
 /// flowchart LR
-///     Created -- await consensus --> Accepted
-///     Created -- await consensus --> Rejected
-///     Created -- Periodically submit --> Created
-///     Created -- Error on submit --> Rejected
+///     Created -- tx is accepted by consensus --> Accepted
+///     Created -- tx is rejected on submission --> Rejected
 /// ```
 #[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
 pub enum TxSubmissionStates {
     /// The transaction has been created and potentially already been submitted,
     /// but no rejection or acceptance happened so far
-    Created {
-        // TODO: put into wrapper struct
-        /// We need this filed to be able to submit multiple transactions in one
-        /// operation.
-        txid: TransactionId,
-        tx: Transaction,
-        next_submission: SystemTime,
```

Comment:
Removing this is good, explicitly modeling a state transition from created to created after some time was a mistake (spams the inactive state log).